### PR TITLE
HPCC-21297 Create System Servers Widget

### PR DESCRIPTION
**Pull request by miguelvazq**
Fixes: HPCC-21297

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/src/Utility.ts [hunk 1]
```diff
@@ -659,6 +659,9 @@ export function resolve(hpccWidget, callback) {
         case "SourceFilesWidget":
             require(["hpcc/SourceFilesWidget"], doLoad);
             break;
+        case "SystemServersQueryWidget":
+            require(["hpcc/SystemServersQueryWidget"], doLoad);
+        break;
```
Comment (GordonSmith) [inline esp/src/src/Utility.ts]: Indentation (minor)

File: esp/src/eclwatch/SystemServersQueryWidget.js [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+                this.addChild(retVal, "last");
+            }
+            return retVal;
+        },
+
+        ensureMIPane: function (id, params) {
+            id = this.createChildTabID(id);
+            var retVal = registry.byId(id);
+            if (!retVal) {
+                retVal = new PreflightDetailsWidget({
```
Comment (GordonSmith) [inline esp/src/eclwatch/SystemServersQueryWidget.js]: Where is PreflightDetailsWidget defined?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith Updated.

**Review by GordonSmith — APPROVED** [review]
Looks ok.

**Comment by richardkchapman** [comment]
Is this something that should go in 7.6 ?

**Comment by miguelvazq** [comment]
It could potentially but, the other two components went into 7.4 and I believe it was mentioned it could go into a 7.4.6 / 7.4.8 if not 7.6 is fine. This widget will contain the legacy System Servers alongside the new ECL Watch improvement so it does not break any functionality.

**Comment by richardkchapman** [comment]
@GordonSmith your call

**Comment by GordonSmith** [comment]
7.6.x would be my preference.

**Comment by miguelvazq** [comment]
> 7.6.x would be my preference?

I'm OK with that.

**Comment by GordonSmith** [comment]
@miguelvazq can you re-target this PR?

**Comment by miguelvazq** [comment]
@GordonSmith Re-targeted.

**Pull request merged**